mesonbuild/compilers/cpp.py

```diff
@@ -764,11 +764,11 @@ class VisualStudioLikeCPPCompilerMixin(CompilerMixinBase):
 
     def get_option_link_args(self, target: 'BuildTarget', env: 'Environment', subproject: T.Optional[str] = None) -> T.List[str]:
         # need a typeddict for this
-        key = self.form_compileropt_key('winlibs')
+        key = self.form_compileropt_key('winlibs').evolve(subproject=subproject)
         if target:
             value = env.coredata.get_option_for_target(target, key)
```

Comment:
Does the fact that you added the subproject changes the behavior here^

Reply:
It shouldn't, although this function is already fragile. The only way the behavior would change is if the subproject argument and the target.subproject are different

Reply:
Probably... We should not pass in subproject and get the subproject from the target.